kolibri/plugins/coach/assets/src/composables/useExerciseResources.js

```diff
@@ -0,0 +1,213 @@
+import { get, set } from '@vueuse/core';
+import { onMounted, computed, ref, provide, inject } from 'kolibri.lib.vueCompositionApi';
+import { ContentNodeResource, ChannelResource } from 'kolibri.resources';
+import { ContentNodeKinds } from 'kolibri.coreVue.vuex.constants';
+
+/**
+ * @module useExerciseResources
+ * @description A system for fetching exercises and massaging them into the format we need to
+ * display them in the app. Providing annotations, methods for pagination, and other utilities
+ * related to the navigation of topics and their exercise descendants.
+ *
+ * Set the topicId for which you want to view the descendants and then the computed properties
+ * will reflect that topic's descendants automatically (with a loading state set accordingly).
+ *
+ * This is intended for use in a use case where the user is navigating the content tree in order
+ * to select contents.
+ *
+ * This module also reflects the pattern used by the ContentNodeResource.fetchTree method which
+ * provides metadata with which we can easily implement lazy loading of content nodes. In short, we
+ * get a `more` object which gives us the paramaters we need to use to fetch the next batch of nodes
+ * if they're available under the current topic.
+ *
+ */
+export default function useExerciseResources() {
+  /** @type {ref<string>} The topicId for which we will reflect the child nodes */
+  const _currentTopicId = ref(null);
```

Comment:
I think we could simplify some of the logic here by accepting slight coupling between the VueRouter route and this module. It would be relatively simple to turn this into a computed property that reads directly from the route state, see here for an example: https://github.com/learningequality/kolibri/blob/release-v0.16.x/kolibri/plugins/learn/assets/src/composables/useSearch.js#L153

(in future versions, VueRouter directly exposes composable state, so we could migrate to using that).